helpers/preview/bin/preview-rst

```diff
@@ -0,0 +1,313 @@
+# information regarding copyright ownership.  Crate licenses this file to you
+# under the Apache License, Version 2.0 (the "License"); you may not use this
+# file except in compliance with the License.  You may obtain a copy of the
+# License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+# However, if you have executed another commercial license agreement with Crate
+# these terms will supersede the license and you may use the software solely
+# pursuant to the terms of the relevant commercial agreement.
+
+
+# =============================================================================
+# Preview an RST file
+# =============================================================================
+
+# This script can be used to generate an HTML preview of any RST file you may
+# be editing.
+#
+# Usage:
+#
+#    preview-rst <RST_FILE>
+#
+# This script is not intended to be run by hand. It has been written with the
+# purpose of being integrated into a desktop editing environment. Integration
+# is possible if you are using an editor that allows you to extend its
+# functionality with arbitrary scripting.
+#
+# In particular, you may get the most use out of this script if you bind it to
+# a keyboard shortcut like <Command-P> (for "preview"). This would likely
+# involve overriding the "Print" keyboard shortcut of your editor, which,
+# depending on your situation, may not be too important.
+#
+# This script has two modalities:
+#
+# - Running this script on RST files in a Sphinx project should open a new
+#   browser window to the corresponding HTML page.
+#
+#   If you are editing lots of files, this can save a significant amount of
+#   time because it does not require you to navigate to each page manually to
+#   check your work.
+#
+# - Running this script on a standalone file (e.g., `CHANGES.rst`) will build
+#   an HTML version and open a new browser window to display it.
+#
+#   This is helpful because without pushing a file to GitHub (or elsewhere)
+#   there is no easy way to preview your work. Indeed, lots of errors may creep
+#   in because of this limitation.
+#
+# For standalone files, there are two strategies currently implemented:
+#
+# - The basic strategy (`display_standalone`) uses pandoc to generate a simple
+#   HTML file. This file has no styling whatsoever and, as a result, the result
+#   may be hard to read. This strategy has been disabled in favor of the Sphinx
+#   strategy (see next item).
+#
+# - The Sphinx strategy (`display_standalone_sphinx`) creates a temporary
+#   Sphinx project for your RST file, starts `sphinx-autobuild`, and opens a
+#   new browser window.
```

Comment:
Maybe swap those sections, have the `display_standalone_sphinx` strategy come first, and say on the second one, like:

> There is also a more basic `display_standalone` strategy, which uses Pandoc to....

Reply:
I would prefer not to do this. generally, in code, the simple stuff comes first, and then you build on that as you progress through the file

in this case, the more basic strategy was built first. it makes sense to me that the more complex strategy comes after

I will probably remove the basic strategy after a bit of battle testing of the more advanced one